### PR TITLE
[PLAYER-2947] dev-epam to mater

### PR DESCRIPTION
**Pull request by KseniiaSokolovaEpam**
New function added that sets audio track in accordance with specified trackID.
Unit tests added.
Unit tests were run without errors.

### CODE REVIEW

File: src/main/js/main_html5.js [hunk 1]
```diff
@@ -873,11 +875,37 @@ require("../../../html5-common/js/utils/environment.js");
             lang: audioTracks[index].language,
             enabled: audioTracks[index].enabled
           };
+          if (audioTracks[index].enabled) {
```
Comment (jpfranco) [inline src/main/js/main_html5.js]: Do we need a null check for `audioTracks[index]`?

File: src/main/js/main_html5.js [hunk 2]
```diff
@@ -873,11 +875,37 @@ require("../../../html5-common/js/utils/environment.js");
...
-      return [];
+      return audioTrackList;
+    };
+
+    /**
+     * Sets the audio track to the ID specified by trackID
+     * @public
+     * @method OoyalaVideoWrapper#setAudio
+     * @param {String} trackId - the ID of the audio track to activate
+     * @returns {Boolean} true - if new audio track was set; false is in otherwise;
```
Comment (jpfranco) [inline src/main/js/main_html5.js]: Small type, could be just "false otherwise"

File: tests/unit/main_html5/wrapper-api-tests.js [hunk 3]
```diff
@@ -1126,6 +1126,16 @@ describe('main_html5 wrapper tests', function () {
...
 
+  it('setAudio should set the audio by id and return true', function(){
+    element.currentAudioId = "0";
+    element.audioTracks = [
+      {id: "0", kind: "main", label: "eng", language: "eng", enabled: true},
+      {id: "1", kind: "main", label: "ger", language: "ger", enabled: false}
+    ];
+    var returns = wrapper.setVideoUrl("1");
+    expect(returns).to.be(true);
+  });
```
Comment (jpfranco) [inline tests/unit/main_html5/wrapper-api-tests.js]: Can we also check that the current audio track is updated properly?

File: src/main/js/main_html5.js [hunk 4]
```diff
@@ -873,11 +875,37 @@ require("../../../html5-common/js/utils/environment.js");
...
         }
-        return audioTrackList;
       }
-      return [];
+      return audioTrackList;
+    };
+
+    /**
+     * Sets the audio track to the ID specified by trackID
+     * @public
```
Comment (aeng7) [inline src/main/js/main_html5.js]: Can we update the video plugin template with this new public function? The template acts as a form of documentation for video plugin developers.

File: tests/unit/main_html5/wrapper-api-tests.js (hunk 3 above)
Comment (aeng7) [inline tests/unit/main_html5/wrapper-api-tests.js]: Can we add a unit test to check that passing in an invalid track id does not cause any issues?

File: src/main/js/main_html5.js (hunk 1 above)
Comment (KseniiaSokolovaEpam) [reply src/main/js/main_html5.js]: No, it is loop. Index is index of a track, the track can be null.

File: src/main/js/main_html5.js (hunk 2 above)
Comment (KseniiaSokolovaEpam) [reply src/main/js/main_html5.js]: \+

File: src/main/js/main_html5.js (hunk 1 above)
Comment (jpfranco) [reply src/main/js/main_html5.js]: I know it's unlikely, but it's possible to have an array like:
```
[{}, null, {}, {}]
```
The check above would throw an error in this case. Even though the tracks are not supposed to contain nulls, I assume they are coming from external sources (i.e. Bitmovin, the video element, etc.) so we shouldn't trust that they will always have the expected format.

File: tests/unit/main_html5/wrapper-api-tests.js (hunk 3 above)
Comment (KseniiaSokolovaEpam) [reply tests/unit/main_html5/wrapper-api-tests.js]: \+

File: src/main/js/main_html5.js (hunk 4 above)
Comment (KseniiaSokolovaEpam) [reply src/main/js/main_html5.js]: \+

File: src/main/js/main_html5.js [hunk 5]
```diff
@@ -889,20 +890,25 @@ require("../../../html5-common/js/utils/environment.js");
...
-        if (typeof newAudioTrack !== 'undefined') { //if trackId is correct and the audio exists
-          var prevAudioTrack = audioTracks.getTrackById(currentAudioId);
-          if (typeof newAudioTrack !== 'undefined') { //if currentAudioId is correct and the audio exists
-            prevAudioTrack.enabled = false; //the audio is not active anymore
+      if (this.currentAudioId !== trackId) {
+        var audioTracks = _video.audioTracks;
+        if (audioTracks && audioTracks.length) { //if audioTracks exist
+
+          var newAudioTrack = audioTracks.getTrackById(trackId);
+          if (newAudioTrack && typeof newAudioTrack !== 'undefined') { //if trackId is correct and the audio exists
```
Comment (jpfranco) [inline src/main/js/main_html5.js]: I think we could do a simpler check here:
``` if (newAudioTrack)```
`newAudioTrack` will be falsy if it's undefined, so the second check might be redundant. Same comment for the other checks in this function.

File: src/template.js [hunk 6]
```diff
@@ -359,6 +359,23 @@ require("../html5-common/js/utils/constants.js");
...
+    /**
+     * Gets the list of all available audio tracks
+     * @public
+     * @method TemplateVideoWrapper#getAvailableAudio
+     * @returns {Array} - an array of all available audio tracks
+     */
+    this.getAvailableAudio = function() {};
+
+    /**
+     * ets the audio track to the ID specified by trackID
```
Comment (aeng7) [inline src/template.js]: Missing G in "Gets"

**Comment by aeng7** [comment]
LGTM. Please check my comment for the template jsdoc so it can be fixed in a future PR.

File: src/main/js/main_html5.js (hunk 5 above)
Comment (KseniiaSokolovaEpam) [reply src/main/js/main_html5.js]: \+

**Pull request merged**